extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -8,45 +8,35 @@ import * as vscode from 'vscode';
 import * as path from 'path';
 import * as fs from 'fs';
 import * as yaml from 'js-yaml';
+import * as fg from 'fast-glob';
 import { BookTreeItem, BookTreeItemType } from './bookTreeItem';
 import * as nls from 'vscode-nls';
 const localize = nls.loadMessageBundle();
 
-
 export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeItem>, azdata.nb.NavigationProvider {
 	readonly providerId: string = 'BookNavigator';
 
 	private _onDidChangeTreeData: vscode.EventEmitter<BookTreeItem | undefined> = new vscode.EventEmitter<BookTreeItem | undefined>();
 	readonly onDidChangeTreeData: vscode.Event<BookTreeItem | undefined> = this._onDidChangeTreeData.event;
-	private _tableOfContentsPath: string[];
+	private _tableOfContentPaths: string[] = [];
 	private _allNotebooks = new Map<string, BookTreeItem>();
 	private _extensionContext: vscode.ExtensionContext;
 	private _throttleTimer: any;
 	private _resource: string;
 
 	constructor(workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext) {
 		let workspacePaths: string[] = workspaceFolders.map(a => a.uri.fsPath);
-		this._tableOfContentsPath = this.getTableOfContentFiles(workspacePaths);
-		let bookOpened: boolean = this._tableOfContentsPath && this._tableOfContentsPath.length > 0;
-		vscode.commands.executeCommand('setContext', 'bookOpened', bookOpened);
+		this.getTableOfContentFiles(workspacePaths);
```

Comment:
Please add a `.then(() => undefined, (err) => // log error message` after calling this promise, or any failure will be marked as an unhandled promise rejection. You may be able to use `.catch` instead (can't remember if that's on Promise or Thenable)

Reply:
Added!